src/asmcnc/apps/drywall_cutter_app/widget_drywall_shape_display.py

```diff
@@ -552,7 +553,8 @@ def select_toolpath(self, shape, toolpath, rotation):
     def text_input_change(self, instance, *args):
         # On startup it seems to call this function and set everything to 0, so check that drywall app is open
         # Also check that the input matches a valid positive float, to allow user to correct mistakes
-        if self.sm.current == 'drywall_cutter' and re.match(r'^\d*\.\d+$', instance.text):
+        if ((self.dwt_config.app_type == config_options.AppType.SHAPES or
```

Comment:
What other app type would it be? Is this really the same check as before?

Reply:
For some reason, the check doesn't pass with the previous version of the code. If you don't click off the text input before pressing go, the new value doesn't get saved. With this, it does get saved. Again, not sure what causes this...

Reply:
I'm confused why even check if app_type is SHAPES or DWT though? There is no other case. app_type isn't an indicator of the current app visible, only which machine version it is really.

Reply:
I assume because go button takes you to a different screen and the sm has already updated? Not sure but I think you might be able to schedule the call for before the next frame which might solve it?

Reply:
It works fine the way it was. just tested it. The new code would only do something like "if True and ..."